RecoMuon/MuonRechitClusterProducer/plugins/RechitClusterProducer.cc

```diff
@@ -0,0 +1,268 @@
+
+typedef std::vector<reco::MuonRecHitCluster> RecHitClusterCollection;
+
+template <typename Trait>
+class RechitClusterProducerT : public edm::global::EDProducer<> {
+  typedef typename Trait::RecHitRef RechitRef;
+  typedef typename Trait::RecHitRefVector RecHitRefVector;
+
+public:
+  explicit RechitClusterProducerT(const edm::ParameterSet&);
+  ~RechitClusterProducerT() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+  const edm::ESGetToken<typename Trait::GeometryType, MuonGeometryRecord> GeometryToken_;
+  edm::EDGetTokenT<typename Trait::InputType> InputToken_;
+  typedef std::vector<reco::MuonRecHitCluster> RecHitClusterCollection;
+
+  const double rParam_;      // distance paramter
+  const int nRechitMin_;     // min number of rechits
+  const int nStationThres_;  // min number of rechits to count towards nStation
+};
+
+template <typename Trait>
+RechitClusterProducerT<Trait>::RechitClusterProducerT(const edm::ParameterSet& iConfig)
+    : GeometryToken_(esConsumes<typename Trait::GeometryType, MuonGeometryRecord>()),
+      InputToken_(consumes<typename Trait::InputType>(iConfig.getParameter<edm::InputTag>("recHitLabel"))),
+      rParam_(iConfig.getParameter<double>("rParam")),
+      nRechitMin_(iConfig.getParameter<int>("nRechitMin")),
+      nStationThres_(iConfig.getParameter<int>("nStationThres")) {
+  produces<RecHitClusterCollection>();
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+template <typename Trait>
+void RechitClusterProducerT<Trait>::produce(edm::StreamID, edm::Event& ev, const edm::EventSetup& iSetup) const {
+  auto const& geo = iSetup.getData(GeometryToken_);
+  auto const& rechits = ev.get(InputToken_);
+
+  std::vector<fastjet::PseudoJet> fjInput;
+  RecHitRefVector inputs_;
+
+  fastjet::JetDefinition jet_def(fastjet::cambridge_algorithm, rParam_);
+
+  inputs_.clear();
+  fjInput.clear();
+
+  Trait RecHitTrait;
+
+  int recIt = 0;
+  for (auto const& rechit : rechits) {
+    LocalPoint RecHitLocalPosition = rechit.localPosition();
+    auto detid = RecHitTrait.detid(rechit);
+    auto thischamber = geo.chamber(detid);
+    if (thischamber) {
+      GlobalPoint globalPosition = thischamber->toGlobal(RecHitLocalPosition);
+      float x = globalPosition.x();
+      float y = globalPosition.y();
+      float z = globalPosition.z();
+      RechitRef ref = RechitRef(&rechits, recIt);
+      inputs_.push_back(ref);
+      fjInput.push_back(fastjet::PseudoJet(x, y, z, globalPosition.mag()));
+      fjInput.back().set_user_index(recIt);
+    }
+    recIt++;
+  }
+  fastjet::ClusterSequence clus_seq(fjInput, jet_def);
+
+  //keep all the clusters
+  double ptmin = 0.0;
+  std::vector<fastjet::PseudoJet> fjJets = fastjet::sorted_by_pt(clus_seq.inclusive_jets(ptmin));
```

Comment:
according to https://github.com/cms-sw/cmssw/pull/35836#discussion_r753352411 sorting is not necessary
```suggestion
  std::vector<fastjet::PseudoJet> fjJets = clus_seq.inclusive_jets(ptmin);
```

Reply:
Removed